core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -509,25 +513,30 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, long offset,
       BlockReader blockReader = null;
       // timings
       long openMs = -1;
+      long startTransferMs = -1;
       long transferMs = -1;
       long startMs = System.currentTimeMillis();
       try {
         openBlock(context);
         openMs = System.currentTimeMillis() - startMs;
         blockReader = context.getBlockReader();
         Preconditions.checkState(blockReader != null);
-        ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(len, len);
-        try {
-          long startTransferMs = System.currentTimeMillis();
-          while (buf.writableBytes() > 0 && blockReader.transferTo(buf) != -1) {
+        startTransferMs = System.currentTimeMillis();
+        if (IS_READER_BUFFER_POOLED) {
+          ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(len, len);
+          try {
+            while (buf.writableBytes() > 0 && blockReader.transferTo(buf) != -1) {
```

Comment:
Does the efficiency depend on the different implementations of data readers? Or It's just the difference between Pooled or unpooled buffer.

Reply:
Good question! I think the efficiency depend on both.  If the reader returns a direct or mmap buffer (current local file block reader return a mmap buffer), I think using the unpooled wrapped buffer might be better.  For the other cases, I have no idea so far, we might need to perform some tests with the community users.